server/index.js

```diff
@@ -1,4 +1,4 @@
 const App = require('./server.js');
 
 // eslint-disable-next-line no-unused-expressions
-App.listen(process.env.PORT, () => console.log(`Server is listening on port ${process.env.PORT}`));
+App.listen(process.env.PORT || 5004, () => console.log(`Server is listening on port ${process.env.PORT}`));
```

Comment:
this seems useful but i have no idea what environment variable are for or where they come from.

Reply:
environment variables are how you pass along all the secret information like an API key or your mongo passwords. If you wanted to be able to change the port(or map it to another or something) you could inject the PORT variable to the environment where it will be run, then it will show up as process.env.PORT. Since your configs or .envs are gitignored, somewhere you'll need to pass a real password to them...not sure how to do this on ec2 yet though